fix: Kilt runtime 1.12.0 locks

Ad96el opened the pull request:
# fixes [3110](https://github.com/KILTprotocol/ticket/issues/3110)

This draft PR introduces support for both runtime versions. The plan is to phase out V1 support in the upcoming runtime release. 

During testing, I observed that creating a w3n is currently unsuccessful, with no error message displayed in the console. Once the underlying issue causing this error is addressed, the draft PR will be transitioned into a regular PR.

@rflechtner Do you maybe by chance know what the issue can be?

rflechtner commented:
is that peregrine or peregrine-stg?

Ad96el commented:
peregrine-stg is updated yet. The endpoint is: `wss://kilt-peregrine-stg-para-rpc.dwellir.com`.

rflechtner commented:
the fix was too much of a point solution, you didn't fix the other uses of transformBalance. I pushed a proposal that makes this function accept both variants. Feel free to revert and adapt if you don't like it.

Ad96el commented:
Yes… Stupid me… I didn't check whether the `tranformBalance` function is used somewhere else. My fault - sorry :raised_hands: .


I tested everything, and it seems all functionalities are working. I am happy with that solution.

I renamed the `locked` variable to `frozen` in order to be closer to the substrate framework, and removed some unnecessary fields in the tests. 

Do you want to address the TODO in this PR? I would say we should tackle that in a different one.

rflechtner commented:
> I renamed the `locked` variable to `frozen` in order to be closer to the substrate framework, and removed some unnecessary fields in the tests. 

Okay great!

> Do you want to address the TODO in this PR? I would say we should tackle that in a different one. 

That's #2380 ;)

rflechtner commented on src/utilities/balanceChanges/balanceChanges.ts:
```diff
@@ -3,30 +3,27 @@ import { ConfigService } from '@kiltprotocol/sdk-js';
-
 export interface BalanceChange {
   address: string;
   balances: {
```
We could define this type via `TransformedBalances` instead of defining this structure once again

rflechtner reviewed (approved):
Lgtm!
